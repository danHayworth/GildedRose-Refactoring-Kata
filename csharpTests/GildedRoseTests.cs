using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using NUnit.Framework;
using System.Collections.Generic;

namespace csharp.Tests
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    [TestClass()]
    public class GildedRoseTests
    {
        [TestMethod()]
        public void GildedRoseTest()
        {
            IList<Item> items = new List<Item>(Program.Items);
            GildedRose app = new GildedRose(items);
            app.UpdateQuality();
            Approvals.VerifyAll<Item>(items, label: "");           
        }

        [TestMethod()]
        public void UpdateQualityTest()
        {
            CombinationApprovals.VerifyAllCombinations(
                getNames,
                new string[] { },
                new int[] { },
                new int[] { }
                );
        }
        private string getNames(string name, int sellIn, int quality)
        {           
            IList<Item> newItems = new List<Item>() { new Item { Name = name, Quality = quality, SellIn = sellIn } };
            GildedRose gilded = new GildedRose(newItems);
            gilded.UpdateQuality();
            return newItems[0].ToString();
            
        }
    }
}