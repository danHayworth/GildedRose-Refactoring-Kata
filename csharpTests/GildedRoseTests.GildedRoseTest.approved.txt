[0] = +5 Dexterity Vest, 9, 19
[1] = Aged Brie, 1, 1
[2] = Elixir of the Mongoose, 4, 6
[3] = Sulfuras, Hand of Ragnaros, 0, 80
[4] = Sulfuras, Hand of Ragnaros, -1, 80
[5] = Backstage passes to a TAFKAL80ETC concert, 14, 21
[6] = Backstage passes to a TAFKAL80ETC concert, 9, 50
[7] = Backstage passes to a TAFKAL80ETC concert, 4, 50
[8] = Conjured Mana Cake, 2, 5
